### DISCUSSION TITLE
How to run content script without page refresh after extension install?

### DISCUSSION THREAD

**Discussion by Abhinav-birdeye** [crxjs/chrome-extension-tools#899] [Q&A]
Hey everyone, 
I'm grateful for this project, it makes building extensions a breeze. 

## The problem
I have a content script (UI) defined in the manifest (index.tsx). When the extension is installed, I want the content script to run automatically without needing a page refresh. How can this be done? 

## What has been tried
If the answer is to register or execute content script from the background worker, I haven't been able to do that. Just a slew of bugs, and ambiguity on the way to import the TSX file. 

I am just a beginner here, trying to figure things out. Thank you :)

**Comment by Toumash** [comment]
Try content scripts - they work right after installation - you need to refresh the page using something like [`chrome.tabs.reload(  tabId?: number,  reloadProperties?: object,  callback?: function)`](https://developer.chrome.com/docs/extensions/reference/api/tabs#method-reload)

If you really need to execute a script [onInstsall ](https://developer.chrome.com/docs/extensions/reference/api/management#event-onInstalled)from a service_worker try "dynamic content scripts" from [the jacks medium article](https://dev.to/jacksteamdev/advanced-config-for-rpce-3966)
